### PR TITLE
Master gpu trace

### PR DESCRIPTION
**Pull request by dejangrubisic**
This is the most current version of master-gpu-trace. It contains logic for setting a number of how many GPU streams data are managed by each helper thread. By default, this number is set to 65536 which effectively creates only one helper thread. 

Besides this, we added register-like control-knobs logic that enables registering environment variables for only sample sources that we are using.

### CODE REVIEW

File: Makefile.in [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# Makefile.in generated by automake 1.15.1 from Makefile.am.
+# Makefile.in generated by automake 1.16.1 from Makefile.am.
```
Comment (Jokeren) [inline Makefile.in:1]: I guess we should be stick to 1.15.1?

File: src/include/hpctoolkit-config.h.in~ [hunk 2]
```diff
@@ -0,0 +1,223 @@
+/* src/include/hpctoolkit-config.h.in.  Generated from configure.ac by autoheader.  */
```
Comment (Jokeren) [inline src/include/hpctoolkit-config.h.in~:1]: Is this file necessary?

File: src/tool/hpcrun/control-knob.c [hunk 3]
```diff
@@ -1,101 +1,135 @@
...
-char *
-control_knob_value_get(control_category c)
+void
+control_knob_init()
 {
-  if (c < HPCRUN_NUM_CONTROL_KNOBS) {
-    return control_knobs[c].value;
-  }
-  return NULL;
+	control_knob_default_register();
```
Comment (Jokeren) [inline src/tool/hpcrun/control-knob.c:61]: It would be better if you can change the indentation space from four to two.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+/*
```
Comment (Jokeren) [inline .gitignore:1]: Please remove this file from the commit

File: src/include/hpctoolkit-config.h.in~ (hunk 2 above)
Comment (dejangrubisic) [reply src/include/hpctoolkit-config.h.in~:1]: Probably not, I will remove it

File: src/tool/hpcrun/control-knob.c (hunk 3 above)
Comment (dejangrubisic) [reply src/tool/hpcrun/control-knob.c:61]: I would, but in my editor it is already on two. I am not sure why it makes it 4 on git. I could probably change indentation from llnl.

File: .gitignore (hunk 4 above)
Comment (dejangrubisic) [reply .gitignore:1]: No problem, I taught .gitignore shouldn't commit by default.

File: Makefile.in (hunk 1 above)
Comment (jmellorcrummey) [reply Makefile.in]: dejan needs to be using 1.15.1, which is our standard.

**Comment by jmellorcrummey** [comment]
@dejangrubisic @mwkrentel I don't understand the diffs in this pull request. yesterday, I merged master into develop. develop contains mark's changes to parallelism for hpcstruct. specifically, it uses the variable opts.jobs_struct. dejan merged this code into his dejangrubisic:master-gpu-trace branch as well, so it also uses opts.jobs_struct.  nevertheless, the file differences that I see in my browser show the following, where the LHS is develop and the RHS is dejangrubisic:master-gpu-trace. the LHS shows opts.jobs instead of opts.jobs_struct. am I misunderstanding something?

```

 @@ -600,7 +600,7 @@ makeStructure(string filename,
       cout << "file:  " << elfFile->getFileName() << "\n"		       cout << "file:  " << elfFile->getFileName() << "\n"
 	   << "symtab threads: " << opts.jobs_symtab		 	   << "symtab threads: " << opts.jobs_symtab
 	   << "  parse: " << opts.jobs_parse		 	   << "  parse: " << opts.jobs_parse
 	   << "  struct: " << opts.jobs<< "\n\n";		 	   << "  struct: " << opts.jobs_struct << "\n\n";
       printTime("init:  ", &tv_init, &ru_init, &tv_init, &ru_init);		       printTime("init:  ", &tv_init, &ru_init, &tv_init, &ru_init);
     }		     }


```

**Comment by mwkrentel** [comment]
`jobs_struct` is the correct value for master.  This came in in commit
61c658fab3d3 on July 13.  I don't know what's in Dejan's version.
It would depend on whether my commit is in his branch.

**Comment by blue42u** [comment]
@jmellorcrummey Unless I'm not seeing the code you refer to, that code is also present in `master`. The only reason its here is because of Dejan's editor messing with the whitespace.

It is also dead code in the `rtld-audit` branch, the only effect I've noticed is the addition of `<partial call paths>` from ld.so bits.
